ecs-cli/modules/cli/image/image_app.go

```diff
@@ -306,25 +347,70 @@ func getECRAuth(registryURI string, registryID string,
 	stsClient stsclient.Client, ecrClient ecrclient.Client) (*ecrclient.Auth, error) {
 
 	if registryURI == "" {
-		id, err := getRegistryID(registryID, stsClient)
-		if err != nil {
-			return nil, err
-		}
-		return ecrClient.GetAuthorizationTokenByID(id)
+		return ecrClient.GetAuthorizationTokenByID(registryID)
 	}
 
 	return ecrClient.GetAuthorizationToken(registryURI)
 }
 
-func splitImageName(image string, seperatorRegExp string, format string) (registry string, repository string, tag string,  err error) {
+func splitImageName(image string, seperatorRegExp string, format string) (registry string, repository string, tag string, err error) {
 	re := regexp.MustCompile(
 		`^(?:((?:[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(\-fips)?\.[a-zA-Z0-9\-_]+\.amazonaws\.com(?:\.cn)?)/)?` + // registry uri (Optional)
-		`([0-9a-z\-_/]+)` + // repository
-		`(?:` + seperatorRegExp + `([0-9A-Za-z_.\-:]+))?$`) // tag or sha (Optional)
-		matches := re.FindStringSubmatch(image)
-		if len(matches) == 0 {
-			return "", "", "", fmt.Errorf("Please specify the image name in the correct format [%s]", format)
-		}
+			`([0-9a-z\-_/]+)` + // repository
+			`(?:` + seperatorRegExp + `([0-9A-Za-z_.\-:]+))?$`) // tag or sha (Optional)
+	matches := re.FindStringSubmatch(image)
+	if len(matches) == 0 {
+		return "", "", "", fmt.Errorf("Please specify the image name in the correct format [%s]", format)
+	}
+
+	return matches[1], matches[3], matches[4], nil
+}
+
+func getRegistryIDFromURI(registryURI string) (string, error) {
+	re := regexp.MustCompile(`^([a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(\-fips)?\.[a-zA-Z0-9\-_]+\.amazonaws(\.com)?(\.cn)?`)
+	matches := re.FindStringSubmatch(registryURI)
+	if len(matches) < 2 {
+		return "", fmt.Errorf("Could not parse account ID from registry URI; URI=%s", registryURI)
+	}
+	return matches[1], nil
+}
+
+func tagRepo(repoARN string, tags map[string]*string, taggingClient tagging.Client) error {
+	input := &resourcegroupstaggingapi.TagResourcesInput{
+		ResourceARNList: aws.StringSlice([]string{
+			repoARN,
+		}),
+		Tags: tags,
+	}
+	output, err := taggingClient.TagResources(input)
+	if err != nil {
+		return err
+	}
 
-		return matches[1], matches[3], matches[4], nil
+	for resource, info := range output.FailedResourcesMap {
+		return fmt.Errorf("Failed to tag respository %s; error=%s", resource, *info.ErrorMessage)
 	}
+	return nil
+}
+
+// arn:aws:ecr:region:account-id:repository/repository-name
+func getRepoARN(region, accountID, repository string) string {
+	repoARN := arn.ARN{
+		Partition: getPartition(region),
+		Service:   "ecr",
+		Region:    region,
+		AccountID: accountID,
+		Resource:  "repository/" + repository,
+	}
+	return repoARN.String()
+}
+
+func getPartition(region string) string {
```

Comment:
Getting the partition for a specific region is something we do for `registry-creds` as well, and will probably be needed elsewhere in the future. What do you think about pulling this  into `utils` so it can be used across the app?